base/libgit2/oid.jl

```diff
@@ -155,8 +155,21 @@ function GitShortHash(obj::GitObject)
     return sid
 end
 
-Base.hex(id::GitHash) = join([hex(i,2) for i in id.val])
-Base.hex(id::GitShortHash) = hex(id.hash)[1:id.len]
+assert_sixteen(x) =
+    if x == 16
+        x
+    else
+        error("Must be base 16")
+    end
+
+function Base.base(b, id::GitHash)
+    assert_sixteen(b)
+    join([base(b,i,2) for i in id.val])
+end
+function Base.base(b, id::GitShortHash)
```

Comment:
I'm not sure these methods should exist (same applies to the `hex` methods, which I was unaware of). Converting a GitHash to a string does the same thing.

Reply:
Yeah, let's just deprecate `hex` in favour of `string`.